cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceQueryAdmin.java

```diff
@@ -56,7 +58,8 @@ public DefaultNamespaceQueryAdmin(NamespaceStore nsStore) {
    * @throws NamespaceNotFoundException if the requested namespace is not found
    */
   @Override
-  public NamespaceMeta get(Id.Namespace namespaceId) throws Exception {
+  public NamespaceMeta get(Id.Namespace namespaceId) throws NamespaceNotFoundException, IOException,
+    UnauthenticatedException {
```

Comment:
When will it be thrown? I don't see the operation can result in this. Also in this PR I see a lot of signatures has this exception added, why? When will it be thrown? Many of those calls are intra-CDAP service calls, which shouldn't have this exception, isn't it?